File: capstone_project/ecommerce/src/services/user-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { User, UserDTO } from '../interfaces/user';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class UserService {

   private apiUrl = 'http://localhost:9191/user';

  constructor(private http: HttpClient) { }

  updateUser(id: number, userDTO: UserDTO): Observable<User> {
    return this.http.put<User>(`${this.apiUrl}/${id}`, userDTO);
  }

  deleteUser(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  getAllUsers(): Observable<User[]> {
    return this.http.get<User[]>(this.apiUrl);
  }

  getUserById(id: number): Observable<User> {
    return this.http.get<User>(`${this.apiUrl}/${id}`);
  }

  getUserByUsername(username: string): Observable<User> {
    return this.http.get<User>(`${this.apiUrl}/username/${username}`);
  }
  
}

File: capstone_project/userservice/src/main/java/com/wipro/userservice/service/UserService.java
package com.wipro.userservice.service;

import java.util.List;

import com.wipro.userservice.dto.Token;
import com.wipro.userservice.dto.UserDTO;
import com.wipro.userservice.entity.User;

public interface UserService {
	 	User registerUser(UserDTO userDTO);
	    
	 	User updateUser(Long id, UserDTO userDTO);
	    
	 	void deleteUser(Long id);
	    
	 	List<User> getAllUsers();
	    
	 	User getUserById(Long id);
	    
	 	User getUserByUsername(String username);
	    
	 	Token login(UserDTO userDTO);
	 	
	 	User getCurrentUser(String username);

}
